### PR TITLE
Implement X-RateLimit-Precision handling

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [x] Internal code
- [ ] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This implements the upcoming `X-RateLimit-Precision` header. Unfortunately discord doesn't echo this header so we have to wait for it to be released.

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/internal/requests/Requester.java [hunk 1]
```diff
@@ -260,7 +260,8 @@ private void applyBody(Request<?> apiRequest, okhttp3.Request.Builder builder)
     private void applyHeaders(Request<?> apiRequest, okhttp3.Request.Builder builder, boolean authorized)
     {
         builder.header("user-agent", USER_AGENT)
-               .header("accept-encoding", "gzip");
+               .header("accept-encoding", "gzip")
+               .header("x-ratelimit-precision", "milliseconds");
```
Comment (jhgg) [inline src/main/java/net/dv8tion/jda/internal/requests/Requester.java]: `s/milliseconds/millisecond`

**Comment by MinnDevelopment** [comment]
Waiting for this to be deployed so reactions don't send at lower rates.

**Comment by MinnDevelopment** [comment]
We should expect users to have a proper ntp synchronized clock on their system.

**Comment by caneleex** [comment]
Cloned the repo with this branch and it gives me instant NPE while starting the bot on the line 156 (parseDouble). Will add more details later tomorrow.

**Comment by binaryoverload** [comment]
I'd guess that's because the particular headers are only in planning right now and are not in production?

**Comment by caneleex** [comment]
Ooh I see. I'm dumb, thanks.

**Pull request merged**